src/libraries/System.Text.RegularExpressions/gen/RegexAnalyzer.cs

```diff
@@ -0,0 +1,416 @@
+
+        /// <inheritdoc />
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(DiagnosticDescriptors.UseRegexSourceGeneration);
+
+        /// <inheritdoc />
+        public override void Initialize(AnalysisContext context)
+        {
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+            context.EnableConcurrentExecution();
+
+            // Register analysis of calls to the Regex constructors
+            context.RegisterOperationAction(AnalyzeObjectCreation, OperationKind.ObjectCreation);
+
+            // Register analysis of calls to Regex static methods
+            context.RegisterOperationAction(AnalyzeInvocation, OperationKind.Invocation);
+        }
+
+        /// <summary>
+        /// Analyzes an invocation expression to see if the invocation is a call to one of the Regex static methods,
+        /// and checks if they could be using the source generator instead.
+        /// </summary>
+        /// <param name="context">The operation context representing the invocation.</param>
+        private void AnalyzeInvocation(OperationAnalysisContext context)
+        {
+            // Ensure source generator is supported.
+            if (!ProjectSupportsRegexSourceGenerator(context.Operation))
+            {
+                return;
+            }
+
+            // Ensure the invocation is a Regex static method.
+            IInvocationOperation invocationOperation = (IInvocationOperation)context.Operation;
+            IMethodSymbol method = invocationOperation.TargetMethod;
+            if (!method.IsStatic || !SymbolEqualityComparer.Default.Equals(method.ContainingType, context.Compilation.GetTypeByMetadataName(RegexTypeName)))
+            {
+                return;
+            }
+
+            // Depending on the static method being called, we need to save the parameters as properties so that we can save them onto the diagnostic so that the
+            // code fixer can later use that property bag to generate the code fix and emit the RegexGenerator attribute.
+            // Most static methods have the same parameter overloads which are covered by the first if block. Replace static method takes extra parameters so that one
+            // is treated specially.
+            if (method.Name is "IsMatch" or "Match" or "Matches" or "Split" or "Count" or "EnumerateMatches")
+            {
+                for (int i = 1; i < invocationOperation.Arguments.Length; i++)
+                {
+                    // Ensure that all inputs to the static method are constant.
+                    if (!IsConstant(invocationOperation.Arguments[i]))
+                    {
+                        return;
+                    }
+                }
+
+                // Create the property bag.
+                ImmutableDictionary<string, string?> properties = ImmutableDictionary.CreateRange(new[]
```

Comment:
Is there a reason to add this property bag to the diagnostic instead of computing all the required information in the fixer itself? Given the diagnostic location, the fixer can fetch the required syntax node from the location using `root.FindNode(span)` API. You can get the semantic model from the document using `document.GetSemanticModelAsync` API and then use `semanticModel.GetOperation(node)` API to get to the `IInvocationOperation`.

Reply:
NOTE: If you feel your current approach is more maintainable and makes it more easier to understand for anyone new to this code, then it seems fine as well. We generally don't recommend using such a property bag unless needed as you are now adding an additional dependency between the analyzer and fixer internals, which might make it more harder to maintain or understand.